lldb/source/Core/Progress.cpp

```diff
@@ -106,9 +107,15 @@ void Progress::ReportProgress() {
   if (completed < m_prev_completed)
     return; // An overflow in the m_completed counter. Just ignore these events.
 
+  // Change the category bit if we're an internal or external progress.
+  uint32_t progress_category_bit =
+      m_origin == ProgressOrigin::eExternal
+          ? lldb::eBroadcastBitExternalProgressCategory
+          : lldb::eBroadcastBitProgressCategory;
```

Comment:
This is wrong, it will report all progress events as aggregated by category. You'll need to check both `eBroadcastBitProgress` and `eBroadcastBitProgressCategory` and pick the right external variant.

Reply:
I understand we have two categories and two defined bits here, but I'm confused what you mean by 

> You'll need to check both eBroadcastBitProgress and eBroadcastBitProgressCategory

Wouldn't I just want to set both bits (and unset internal), and pass that to the debugger to see if there is a listener for either?